### PR TITLE
Image encoder

### PR DESCRIPTION
**Pull request by NathanielDamours**
- Fix requirements' issue
- Fix typing in MNIST and CIFAR datasets
- Add config param for a future image feature extractor

### CODE REVIEW

File: config/dataset/cifar100.yaml
Comment (benjaminlblanc) [inline config/dataset/cifar100.yaml:1]: Pourquoi changer les dimensions ici?

File: config/dataset/mnist.yaml
Comment (benjaminlblanc) [inline config/dataset/mnist.yaml:1]: Pourquoi changer les valeurs par défaut ici?

File: config/grid_search_override.yaml
Comment (benjaminlblanc) [inline config/grid_search_override.yaml:1]: Pourquoi changer les valeurs par défaut ici?

File: config/training/gpu.yaml
Comment (benjaminlblanc) [inline config/training/gpu.yaml:1]: Pourquoi changer les valeurs par défaut ici?

File: src/data/dataset/cifar100.py
Comment (benjaminlblanc) [inline src/data/dataset/cifar100.py:1]: Pourquoi "np.stack([tensor.detach().cpu().numpy() for tensor in binary_datasets])" au lieu de "binary_datasets"?

File: src/data/dataset/mnist.py
Comment (benjaminlblanc) [inline src/data/dataset/mnist.py:1]: Pourquoi "np.stack([tensor.detach().cpu().numpy() for tensor in binary_datasets])" au lieu de "binary_datasets"?

File: config/dataset/cifar100.yaml
Comment (NathanielDamours) [reply config/dataset/cifar100.yaml:1]: I rolled back the changes

File: config/dataset/mnist.yaml
Comment (NathanielDamours) [reply config/dataset/mnist.yaml:1]: I rolled back the changes

File: config/grid_search_override.yaml
Comment (NathanielDamours) [reply config/grid_search_override.yaml:1]: I rolled back the changes

File: config/training/gpu.yaml
Comment (NathanielDamours) [reply config/training/gpu.yaml:1]: I rolled back the changes

File: src/data/dataset/cifar100.py
Comment (NathanielDamours) [reply src/data/dataset/cifar100.py:1]: To fix the typing. The cache expects a numpy array

File: src/data/dataset/mnist.py
Comment (NathanielDamours) [reply src/data/dataset/mnist.py:1]: Answered there: https://github.com/GRAAL-Research/DeepRM/pull/71#discussion_r2100867047

**Review by benjaminlblanc — APPROVED** [review]
